src/gocam/cli.py

```diff
@@ -476,5 +480,273 @@ def flatten_models(input_file, input_format, output_format, output_file, fields)
+                logger.info(f"Downloaded tarball to {tmp_file.name}")
+                
+                # Extract the tarball
+                logger.info(f"Extracting tarball to {extract_dir}")
+                with tarfile.open(tmp_file.name, 'r:gz') as tar:
+                    tar.extractall(path=extract_dir)
+                    
+                    # Find the extracted directory
+                    extracted_dirs = [d for d in os.listdir(extract_dir) 
+                                    if os.path.isdir(os.path.join(extract_dir, d))]
+                    if extracted_dirs:
+                        return os.path.join(extract_dir, extracted_dirs[0])
+                    else:
+                        return extract_dir
+                        
+            finally:
+                # Clean up the temporary file
+                os.unlink(tmp_file.name)
+
+    def find_json_models(directory: str) -> List[str]:
+        """Find all JSON model files in a directory."""
+        json_files = []
+        for root, dirs, files in os.walk(directory):
+            for file in files:
+                if file.endswith('.json'):
+                    json_files.append(os.path.join(root, file))
+        
+        logger.info(f"Found {len(json_files)} JSON model files")
+        return json_files
+
+    def load_model_from_file(file_path: str) -> Optional[Model]:
+        """Load a GO-CAM model from a Minerva JSON file."""
+        try:
+            with open(file_path, 'r') as f:
+                minerva_dict = json.load(f)
+                # Convert from Minerva format to GO-CAM model
+                model = MinervaWrapper.minerva_object_to_model(minerva_dict)
+                return model
+        except Exception as e:
+            logger.error(f"Failed to load model from {file_path}: {e}")
+            return None
+
+    def translate_to_networkx(model: Model, output_path: str, model_filename: str):
+        """Translate a model to NetworkX format and save as JSON."""
+        try:
+            # Initialize the translator
+            indexer = Indexer()
+            translator = ModelNetworkTranslator(indexer=indexer)
+            
+            # Translate the model
+            json_output = translator.translate_models_to_json([model], include_model_info=True, indent=2)
+            
+            # Save to file
+            output_file = os.path.join(output_path, f"{model_filename}_networkx.json")
+            with open(output_file, 'w') as f:
+                f.write(json_output)
+            
+            logger.debug(f"Saved NetworkX translation to {output_file}")
+            
+        except Exception as e:
+            logger.error(f"Failed to translate {model_filename} to NetworkX: {e}")
+
+    def translate_to_cx2(model: Model, output_path: str, model_filename: str):
+        """Translate a model to CX2 format and save as JSON."""
+        try:
+            # Translate the model
+            cx2_data = model_to_cx2(model)
```

Comment:
Highly recommend using `validate_iquery_gene_symbol_pattern=False` here to cut down on a lot of logging noise. Honestly it's possible we should flip the default of that parameter to `False`.